metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_tag.py

```diff
@@ -0,0 +1,139 @@
+import logging
+from typing import Dict, List, Optional
+
+from datahub.ingestion.source.snowflake.snowflake_config import (
+    SnowflakeV2Config,
+    TagOption,
+)
+from datahub.ingestion.source.snowflake.snowflake_report import SnowflakeV2Report
+from datahub.ingestion.source.snowflake.snowflake_schema import (
+    SnowflakeDataDictionary,
+    SnowflakeTag,
+    _SnowflakeTagCache,
+)
+from datahub.ingestion.source.snowflake.snowflake_utils import SnowflakeCommonMixin
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class SnowflakeTagExtractor(SnowflakeCommonMixin):
+    def __init__(
+        self,
+        config: SnowflakeV2Config,
+        data_dictionary: SnowflakeDataDictionary,
+        report: SnowflakeV2Report,
+    ) -> None:
+        self.config = config
+        self.data_dictionary = data_dictionary
+        self.report = report
+        self.logger = logger
+
+        self.tag_cache: Optional[_SnowflakeTagCache] = None
+
+    def invalidate_cache(self):
```

Comment:
instead of this invalidate cache method, could we make tag_cache a `dict[db_name, _SnowflakeTagCache]`?

generally cache invalidation bookkeeping is easy to get wrong or miss edge cases, so I'd rather make it obviously correct even if it uses slightly more memory